chain/network/src/routing/routing.rs

```diff
@@ -65,7 +65,95 @@ pub enum EdgeType {
     Removed,
 }
 
-pub type Edge = Arc<EdgeInner>;
+#[derive(BorshSerialize, BorshDeserialize, Clone, Debug, PartialEq, Eq)]
+#[cfg_attr(feature = "test_features", derive(Serialize, Deserialize))]
+pub struct Edge(pub Arc<EdgeInner>);
+
+impl Edge {
+    /// Create an addition edge.
+    pub fn new(
+        peer0: PeerId,
+        peer1: PeerId,
+        nonce: u64,
+        signature0: Signature,
+        signature1: Signature,
+    ) -> Self {
+        Edge(Arc::new(EdgeInner::new(peer0, peer1, nonce, signature0, signature1)))
```

Comment:
if this is the 'addition edge' - should we check that nonce is even?

Reply:
I'll add comments in one PR after https://github.com/near/nearcore/pull/5267